corehq/apps/app_manager/templates/app_manager/v2/form_view_base.html

```diff
@@ -211,7 +211,7 @@
 
         gaTrackLink($('.breadcrumb .view-form'), 'App Builder', 'View Form', 'Breadcrumb');
 
-        {% if allow_cloudcare %}
+        {% if can_preview_form and allow_cloudcare %}
```

Comment:
@orangejenny The change is to hide the "Preview Form" link in case user does not have "Edit Data" permission. I tested on V2 and it seems that there is no link for preview form in v2 but it does set the url for link [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/templates/app_manager/v2/form_view_base.html#L248) so the change in v2 seems to be making no difference.